src/clients/c++/library/grpc_client.cc

```diff
@@ -46,8 +48,22 @@ namespace {
 // If context is created on url that has established Channel, then reuse it.
 std::map<std::string, std::shared_ptr<grpc::Channel>> grpc_channel_map_;
 std::mutex grpc_channel_map_mtx_;
+
+void
+ReadFile(const std::string& filename, std::string& data)
+{
+  std::ifstream file(filename.c_str(), std::ios::in);
```

Comment:
Clear 'data' to make sure return empty if no file.
Better checking if filename is empty?